### DISCUSSION TITLE
Error in Log: "unable to write segment: reached maximum segment size"

### DISCUSSION THREAD

**Discussion by madsinatra** [bluenviron/mediamtx#968] [Questions] [answered]
I use rtsp-simple-server to push vpn connected streams from a private network to a static S3 html page:

Cams --> RTSP-Simple-Server --> Nginx for SSL termination --> HTML static page on S3

After 1-2 days of operation I get multiple of those log entries:

`2022/05/30 17:33:01 WAR [HLS] [muxer cam33] unable to write segment: reached maximum segment size
`

I run everything on Ubuntu Jammy and Googling the error message makes me more or less clueless. Does anybody have a hint, what this issue might be?

**Comment by aler9** [comment, marked as answer]
Hello, that happens due to the RAM exhaustion protection mechanism. The server prevents a HLS segment from becoming bigger than a certain size. You can tune that size with the `hlsSegmentMaxSize` parameter.

**Reply by madsinatra** [reply]
Thanks for the hint - what is recommended max allocation per camera stream? In my testbed I have 8 streams - final rollout will have 40 streams. I can push up RAM more or less as needed as I'm running on AWS.

[a comment by an automated account is not shown]
